lib/ex_aws/credentials_ini/file.ex

```diff
@@ -4,12 +4,120 @@ if Code.ensure_loaded?(ConfigParser) do
     @valid_config_keys ~w(
       aws_access_key_id aws_secret_access_key aws_session_token region
       role_arn source_profile credential_source external_id mfa_serial role_session_name credential_process
+      sso_start_url sso_region sso_account_id sso_role_name
     )
 
     def security_credentials(profile_name) do
       shared_credentials = profile_from_shared_credentials(profile_name)
       config_credentials = profile_from_config(profile_name)
-      Map.merge(config_credentials, shared_credentials)
+      merge_credentials(config_credentials, shared_credentials)
+    end
+
+    defp merge_credentials(
+           %{
+             sso_start_url: sso_start_url,
+             sso_account_id: sso_account_id,
+             sso_role_name: sso_role_name
+           } = config_credentials,
+           _shared_credentials
+         ) do
+      case get_sso_role_credentials(sso_start_url, sso_account_id, sso_role_name) do
+        {:ok, sso_creds} -> {:ok, Map.merge(sso_creds, config_credentials)}
+        {:error, _} = err -> err
+      end
+    end
+
+    defp merge_credentials(config_credentials, shared_credentials) do
+      {:ok, Map.merge(config_credentials, shared_credentials)}
+    end
+
+    defp get_sso_role_credentials(sso_start_url, sso_account_id, sso_role_name) do
+      with {_, {:ok, content_raw}} <-
+             {:read, File.read(get_sso_cache_file(sso_start_url))},
+           {_, {:ok, %{"expiresAt" => expires_at} = content}} <-
+             {:decode, Jason.decode(content_raw)},
```

Comment:
Just some personal stylistic opinions:  I'd consider renaming `content`. When I got down to `request_sso_role_credentials(content, sso_account_id, sso_role_name)` i had to look back up a few lines to remember what it was

Reply:
As this function and `request_sso_role_credentials` are currently written, if `content` is missing the fields `"accessToken"` and `"region"` then...

```
{_, {:ok, %{"expiresAt" => expires_at} = content}} <-
  {:decode, Jason.decode(content_raw)},
```
...will succeed. But then call to `request_sso_role_credentials` in...
```
{_, {:ok, sso_creds}} <-
  {:sso_creds, request_sso_role_credentials(content, sso_account_id, sso_role_name)},
```
will panic with pattern match exception.

If `content` is missing the fields `"accessToken"` and `"region"` then I think we'd like 
```
{_, {:ok, %{"expiresAt" => expires_at} = content}} <-
  {:decode, Jason.decode(content_raw)},
```
to fail, causing us to return an `"SSO cache file contains invalid json"` error (similar to how we fail if `"expiresAt"` is missing).

What we can do is change the calls to look like this...

```
{_, {:ok, %{"expiresAt" => expires_at, "accessToken" => access_token, "region" => region} = content}} <-
  {:decode, Jason.decode(content_raw)},
{_, :ok} <-
  {:expiration, check_sso_expiration(expires_at)},
{_, {:ok, sso_creds}} <-
  {:sso_creds, request_sso_role_credentials(access_token, region, sso_account_id, sso_role_name)},
```

Doing this solves two issues:
1. Missing keys in the JSON will now make this function return an `"invalid json"` error instead of raising a pattern match exception.
2. It eliminates the `content` variable.